spinta/manifests/open_api/helpers.py

```diff
@@ -27,74 +32,295 @@ def read_file_data_and_transform_to_json(path: Path) -> dict:
+        """
+        Returns "Property" data type as a string. If type cannot be detected, defaults to "string"
+        """
+        data_type = self.json_schema.get("type", DEFAULT_PROPERTY_DATATYPE)
+
+        basic_types = {
+            "boolean": "boolean",
+            "integer": "integer",
+            "number": "number",
+            "array": "array",
+            "object": "ref",
+        }
+
+        date_time_types = {"date-time": "datetime", "date": "date", "time": "time"}
+
+        if data_type in basic_types:
+            return basic_types[data_type]
+
+        if data_type == "string":
+            if self.json_schema.get("contentEncoding") == "base64":
+                return "binary"
+            if (string_format := self.json_schema.get("format")) in date_time_types:
+                return date_time_types[string_format]
+
+        return DEFAULT_PROPERTY_DATATYPE
+    
+    def get_enums(self, items: list) -> dict:
+        enum = {}
+        for item in items:
+            if isinstance(item,(list, dict)):  # Only handling primitive type enums
+                return {}
+            enum[item] = {"source": item}
+        return enum
+
+    def get_node_schema_dict(self) -> dict:
+        schema = {
+            "type": self.datatype,
+            "title": self.title,
+            "description": self.description,
+            "external": {"name": self.source},
+            "required": self.required,
+        }
+
+        if self.datatype in ["ref", "backref"]:
+            schema["model"] = self.ref.name
+            schema["external"]["prepare"] = Expr("expand")
+
+        if self.enum:
+            schema['enums'] = {"": self.enum}
+
+        return schema
+
+def get_schema_from_unknown_structure(json_dict: dict) -> dict:
+    if isinstance(json_dict, dict):
+        for key, value in json_dict.items():
+            if key == "schema":
+                return value
+            schema = get_schema_from_unknown_structure(value)
+            if schema:
+                return schema
+    return {}
+
+
+def get_schema_from_response(response: dict, root: dict) -> dict:
+    json_schema = {}
+    if "openapi" in root:
+        for content_type, content in response.get("content", {}).items():
+            if content_type == "application/json" or content_type.endswith("+json"):
+                json_schema = content.get("schema", {})
+                break
+    elif "swagger" in root:
+        json_schema = response.get("schema", {})
+    else:
+        json_schema = get_schema_from_unknown_structure(response)
+
+    if SCHEMA_REF_KEY in json_schema:
+        raise NotImplementedFeature(feature="Reading OpenAPI with '$ref' structure")
+    return json_schema
+
+def get_model_schemas(dataset_name: str, resource_name: str, response: dict, root: dict) -> list[dict]:
+    if not (json_schema := get_schema_from_response(response, root)):
+        return []
+
+    if json_schema.get("type") == "array":
```

Comment:
`json_schema.get("type")` re-used, could be extracted to a variable to not be accessed twice